src/js/components/Calendar/Calendar.js

```diff
@@ -481,16 +499,39 @@ const Calendar = forwardRef(
       if (initialFocus === 'days') daysRef.current.focus();
     }, [initialFocus]);
 
+    const handleReference = useCallback(
+      (nextReference) => {
+        setReference(nextReference);
+        if (onReference) onReference(nextReference.toISOString());
+      },
+      [onReference],
+    );
+
     const changeReference = useCallback(
       (nextReference) => {
-        if (betweenDates(nextReference, bounds)) {
-          setReference(nextReference);
-          if (onReference) onReference(nextReference.toISOString());
-        }
+        if (betweenDates(nextReference, bounds)) handleReference(nextReference);
       },
-      [onReference, bounds],
+      [handleReference, bounds],
     );
 
+    const changeCalendarMonth = (messageId, newMonth) => {
+      if (!betweenDates(newMonth, bounds)) handleReference(newMonth);
+      else changeReference(newMonth);
```

Comment:
```suggestion
      handleReference(newMonth);
```
Can we just call `handleReference` here since changeReference does the same check we just did and either way we ultimately call `handleReference` anyway?

Reply:
Yeah, that makes all the sense. Thanks for the tip @jcfilben.